### DISCUSSION TITLE
Celery memory leak in Tribe results in OOM warm restart

### DISCUSSION THREAD

**Discussion by acherla** [StreetLamb/tribe#130] [General]
Celery containers appear to be hitting a memory leak, setting container limits to 5GB or 10GB appears to result in all memory being used by the container even when MAX_WORKERS=1 and when process is completely idle.  This is likely due to a bug in celery version 5.4.0 which is resulting in a memory leak that causes memory usage to increase incrementally until hitting an OOM.  

<img width="518" alt="image" src="https://github.com/user-attachments/assets/427da558-8b63-494c-9e9f-0ef0b2cc1fa2">

**Comment by acherla** [comment]
testing this on python3.11 to see if its only impacting the python 3.10 image

**Reply by StreetLamb** [reply]
Thanks, will take a look. I dont think its a celery issue. The memory spike seems to be happening right when documents are uploaded to the qdrant collection.

**Reply by acherla** [reply]
I would normally agree, but this seems to be occuring even on brand new fresh clusters with no documents.

**Reply by StreetLamb** [reply]
I pushed #133 which helps to alleviate the memory leak issue for me. Let me know if it addresses yours too.

**Reply by acherla** [reply]
Thanks we did the same thing on our side
